### ISSUE TITLE
Array of objects - JSON body as request payload

### ISSUE THREAD

**Issue by aravindvnair99** [apidoc/apidoc#993]
I wanted to show the following as request:

```json
{
  "send": [
    {
      "id": 1,
      "message": "hi"
    },
    {
      "id": 2,
      "message": "bye",
    }
  ]
}
```

The method is `POST` and request payload should be a JSON body only. I'm unable to list it using `@apiBody`. So far I managed to put up an example with this:

```js
* @apiExample {json} JSON sample structure:
{
  "send": [
    {
      "id": 1,
      "message": "hi"
    },
    {
      "id": 2,
      "message": "bye",
    }
  ]
}
```
But when it comes to the sample request, I'm unable to form the array of objects part using `@apiBody`. Is `Object[]` not possible?

```js
* @apiBody {Object[]} send
* @apiBody {Number} send[0].id
* @apiBody {String} send[0].message
```

shows nothing

```js
* @apiBody {Object[]} send
* @apiBody {Number} send.id
* @apiBody {String} send.message
```

shows this:

```json
{
    "send": null,
    "send.id": 0,
    "send.message": ""
}
```

**Comment by NicolasCARPi** [comment]
As per the freshly merged documentation, use this:

~~~
 * @apiBody {Object} [send]
 * @apiBody {Number} [send[id]]
 * @apiBody {String} [send[message]]
 ~~~

**Comment by aravindvnair99** [comment]
> As per the freshly merged documentation, use this:
> 
> ```
>  * @apiBody {Object} [send]
>  * @apiBody {Number} [send[id]]
>  * @apiBody {String} [send[message]]
> ```

Hey @NicolasCARPi!

Thanks for the quick response. I had tried that as well, but that gives this:

```json
{
    "send": {
        "id": 0,
        "message": ""
    }
}
```

It still isn't an array of objects such as:

```json
{
  "send": [
    {
      "id": 1,
      "message": "hi"
    },
    {
      "id": 2,
      "message": "bye",
    }
  ]
}
```

Could you please reopen the issue as it continues to exist?

**Comment by NicolasCARPi** [comment]
Not sure what you want is possible yet.

**Comment by NicolasCARPi** [comment]
You can try the current `dev` branch. See #999 and remove the `Object` params.

**Comment by aravindvnair99** [comment]
> You can try the current `dev` branch. See #999 and remove the `Object` params.

Hey @NicolasCARPi! Thanks for the update. I get the following error though:

```console
❯  npm run docs

> project_name@1.0.0 docs
> rimraf public/docs && apidoc -i api/controllers/ -o public/docs

error: ENOENT: no such file or directory, lstat '/home/username/project_name/node_modules/apidoc/node_modules/bootstrap/dist/css/bootstrap.min.css'
[error] apidoc encountered an error during documentation generation!
```

**Comment by NicolasCARPi** [comment]
Your apidoc is installed globally, right?

**Comment by NicolasCARPi** [comment]
Also, re-reading this issue I see you want array of things, and I didn't look into this yet.

**Comment by aravindvnair99** [comment]
> Your apidoc is installed globally, right?

@NicolasCARPi No, it was locally installed in the project folder. I just tried installing globally and it worked. Is local installation not possible?

> Also, re-reading this issue I see you want array of things, and I didn't look into this yet.

Yeah I wanted to create an array of objects.

# Trials

```js
* @apiBody {Object[]} send
* @apiBody {Number} send[0].id
* @apiBody {String} send[0].message
```
gives me this:

![image](https://user-images.githubusercontent.com/22199259/133206131-8f1a3d71-aa65-4f6a-a0e6-2da57073d592.png)

```js
* @apiBody {Object[]} send
* @apiBody {Number} send.id
* @apiBody {String} send.message
```

gives me this:

![image](https://user-images.githubusercontent.com/22199259/133206481-3b624390-5d4a-4481-b965-ab59081c058a.png)

**Comment by NicolasCARPi** [comment]
> Is local installation not possible?

It should be. I think I still have to sort out the `node_modules` folder nesting. Here it tries to find the things in the `node_modules` folder at the same level of the root of apidoc library files.

**Comment by NicolasCARPi** [comment]
The path issue has been addressed. It now tries to find assets in several folders.

**Comment by NicolasCARPi** [comment]
@chrillep I'm not actively working on apidoc these days. You can help by reviewing #1063 and see if it works for you. I'm open to merging it once it's properly tested.
